src/codegen/codegen_c_visitor.cpp

```diff
@@ -440,11 +440,13 @@ int CodegenCVisitor::position_of_int_var(const std::string& name) const {
  * then it gets printed as an integer. To avoid this, we use below wrapper.
  * If user has provided integer then it gets printed as 1.0 (similar to mod2c
  * and neuron where ".0" is appended). Otherwise we print double variables as
- * they are represented in the mod file by user.
+ * they are represented in the mod file by user. If the value is in scientific
+ * representation (1e+20, 1E-15) then keep it as it is.
  */
 std::string CodegenCVisitor::double_to_string(const std::string& s_value) {
```

Comment:
it is still a little puzzling that double_to_string gets a string and not a double for conversion. Maybe we should find a better name... dunno

Reply:
It's a bit puzzling at first, I agree. I am going to add some more documentation about that to explain why there is a string there.
I cannot think any better name for this tbh :confused:

Reply:
reformat_stringed_double?

Reply:
`format_double_string` and `format_float_string` may be even better?